app/controllers/concerns/collection_scope.rb

```diff
@@ -0,0 +1,48 @@
+module Concerns::CollectionScope
+  extend ActiveSupport::Concern
+
+  private
+
+  def scope
+    raise NotImplementedError, "Please override #scope when you use Concerns::CollectionScope"
+  end
+
+  def default_per_page
+    20
+  end
+
+  def max_per_page
+    250
+  end
+
+  def per_page
+    # allow max_per_page and default_per_page to be nil as well
+    if params[:per_page]
+      [params[:per_page].to_i, max_per_page].compact.min
+    else
+      default_per_page
+    end
+  end
+
+  def search_scope
+    s = params[:q] ? scope.ransack(params[:q]).result(distinct: true) : scope
+    s = s.page(params[:page].to_i).per(per_page) if per_page && per_page >= 0
```

Comment:
does `per_page=0` make sense?

Reply:
That's when you want to know the number of items (I've found myself using that in APIs).

Reply:
if you say so...